cmd/controller/main.go

```diff
@@ -70,3 +86,16 @@ func main() {
 		WithWebhooks(ctx, webhooks.NewWebhooks()...).
 		Start(ctx)
 }
+
+func validateK8sVersion(ctx context.Context, op *operator.Operator) error {
+	serverVersion := lo.Must1(op.VersionProvider.Get(ctx))
+	floatServerVersion := lo.Must1(strconv.ParseFloat(serverVersion, 32))
```

Comment:
This feels _a bit_ tenuous to me just because the version is _technically_ not a float and is a SemVer version. I _do_ think that this logic works fine here since we're always returning `x.y` but if we ever returned `x.y.z`, this would break

Reply:
Maybe there's a way to parse the version and do a version comparison instead? That would hold up over time a little more strongly.